src/devices/RGBLedMatrix/samples/RGBLedMatrix.sample.cs

```diff
@@ -0,0 +1,678 @@
+            scenario = 0;
+            Thread.Sleep(1000);
+            matrix.Dispose();
+        }
+
+        static unsafe void Demo1(RGBLedMatrix matrix)
+        {
+            play = true;
+
+            try
+            {
+                BdfFont font = BdfFont.Load(@"fonts/10x20.bdf");
+                BdfFont font1 = BdfFont.Load(@"fonts/8x13B.bdf");
+                matrix.Fill(0, 0, 0);
+                Thread.Sleep(100);
+
+                int x = matrix.Width - 1;
+                string text = "Hello .NET IoT";
+                int fullTextWidth = text.Length * font.Width;
+
+                while (play)
+                {
+                    matrix.DrawText(x, 0, text, font, 0, 0, 255, 0, 0, 0);
+                    x--;
+                    if (x == -fullTextWidth)
+                        x = matrix.Width - 1;
+
+                    string d = DateTime.Now.ToString("hh:mm:ss");
+                    matrix.DrawText(0, font.Height + 1, d, font1, 128, 128, 0, 0, 0, 0);
+
+                    Thread.Sleep(25);
+                }
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+            }
+        }
+
+        static void Demo2(RGBLedMatrix matrix)
+        {
+            int length = matrix.Width / 4;
+            int height =  matrix.Height / 4;
+
+            play = true;
+            while (play)
+            {
+                matrix.FillRectangle(0, 0, length, length, 255, 0, 0);
+                matrix.FillRectangle(length,     0, length, length, 0,   255, 0);
+                matrix.FillRectangle(2 * length, 0, length, length, 0,   0,   255);
+                matrix.FillRectangle(3 * length, 0, length, length, 255, 255, 0);
+
+                matrix.FillRectangle(0,  height, length, length, 255, 0,   255);
+                matrix.FillRectangle(length,  height, length, length, 255, 255, 255);
+                matrix.FillRectangle(2 * length, height, length, length, 0,   130, 0);
+                matrix.FillRectangle(3 * length, height, length, length, 130, 0,   0);
+
+                matrix.FillRectangle(0,  2 * height, length, length, 0,   0,   128);
+                matrix.FillRectangle(length,  2 * height, length, length, 192, 192, 192);
+                matrix.FillRectangle(2 * length, 2 * height, length, length, 128, 128, 0);
+                matrix.FillRectangle(3 * length, 2 * height, length, length, 128, 128, 128);
+
+                matrix.FillRectangle(0,  3 * height, length, length, 40, 40, 40);
+                matrix.FillRectangle(length,  3 * height, length, length,  80, 80, 80);
+                matrix.FillRectangle(2 * length, 3 * height, length, length,  120,  120,  120);
+                matrix.FillRectangle(3 * length, 3 * height, length, length,  0,  120,  120);
+
+                Thread.Sleep(5000);
+            }
+        }
+
+        private static readonly string s_weatherKey = "Please request a key from openweathermap.org";
```

Comment:
perhaps set this to null, and in Demo3 (and whichever else uses it) add a null check and throw exception with this message

Reply:
I am afraid the exception experience is not that good. I prefer to leave it as it is especially I add a comment in the readme file about that. let me know if you disagree.

Reply:
sounds good to me